iModelJsNodeAddon/presentation/ECPresentationUtils.h

```diff
@@ -46,59 +44,46 @@ struct ECPresentationResult
-        }
-    //! Create a success result with response
-    ECPresentationResult(rapidjson::Value&& successResponse, bool serializeResponse, rapidjson::Document&& diagnostics = rapidjson::Document())
-        : m_status(ECPresentationStatus::Success), m_isJsonCppResponse(false), m_diagnostics(std::move(diagnostics))
-        {
-        successResponse.Swap(m_successResponse);
+    ECPresentationResult(BeJsDocument&& successResponse, bool serializeResponse, rapidjson::Document&& diagnostics = rapidjson::Document())
+        : m_status(ECPresentationStatus::Success), m_successResponse(std::move(successResponse)), m_diagnostics(std::move(diagnostics))
+    {
         if (serializeResponse)
             GetSerializedSuccessResponse();
-        }
+    }
     //! Create a success result with response
-    ECPresentationResult(Json::Value successResponse, bool serializeResponse, rapidjson::Document&& diagnostics = rapidjson::Document())
-        : m_status(ECPresentationStatus::Success), m_jsoncppSuccessResponse(successResponse), m_isJsonCppResponse(true), m_diagnostics(std::move(diagnostics))
-        {
+    ECPresentationResult(BeJsConst successResponse, bool serializeResponse, rapidjson::Document&& diagnostics = rapidjson::Document())
+        : m_status(ECPresentationStatus::Success), m_diagnostics(std::move(diagnostics))
+    {
+        m_successResponse.From(successResponse);
         if (serializeResponse)
             GetSerializedSuccessResponse();
-        }
+    }
     //! Create an error result
     ECPresentationResult(ECPresentationStatus errorCode, Utf8String message, rapidjson::Document&& diagnostics = rapidjson::Document())
         : m_status(errorCode), m_errorMessage(message), m_diagnostics(std::move(diagnostics))
         {}
     //! Move constructor
     ECPresentationResult(ECPresentationResult&& other)
-        : m_status(other.m_status), m_isJsonCppResponse(other.m_isJsonCppResponse), m_errorMessage(std::move(other.m_errorMessage)), m_serializedSuccessResponse(std::move(other.m_serializedSuccessResponse))
+        : m_status(other.m_status), m_errorMessage(std::move(other.m_errorMessage)), m_serializedSuccessResponse(std::move(other.m_serializedSuccessResponse)), m_successResponse(std::move(other.m_successResponse))
         {
         m_diagnostics.Swap(other.m_diagnostics);
-        m_successResponse.Swap(other.m_successResponse);
-        m_jsoncppSuccessResponse.swap(other.m_jsoncppSuccessResponse);
         }
     //! Move assignment
     ECPresentationResult& operator=(ECPresentationResult&& other)
         {
         m_status = other.m_status;
-        m_isJsonCppResponse = other.m_isJsonCppResponse;
         m_errorMessage.swap(other.m_errorMessage);
         m_diagnostics.Swap(other.m_diagnostics);
-        m_successResponse.Swap(other.m_successResponse);
-        m_jsoncppSuccessResponse.swap(other.m_jsoncppSuccessResponse);
+        m_successResponse.From(other.m_successResponse);

```

Comment:
does `From` copy or move?

Reply:
created =operator in BeJsDocument to allow moving

Reply:
updated